### PR TITLE
Add B2B self service application logic

### PR DESCRIPTION
**Pull request by inthirakumaaran**
### Proposed changes in this pull request

- Improve the Application/SP modal to contain b2b self-service property
- Improve Add & Update application logic in DAO
- Add user creation login (need to move to organization core)

### When should this PR be merged

[Please describe any preconditions that need to be addressed before we
can merge this pull request.]

### Follow up actions

[List any possible follow-up actions here; for instance, testing data
migrations, software that we need to install on staging and production
environments.]

-

#### General

- [ ] **Is this PR explained thoroughly?** All code changes must be accounted for in the PR description.
- [ ] **Is the PR labeled correctly?**

#### Functionality

- [ ] **Are all requirements met?** Compare implemented functionality with the requirements specification.
- [ ] **Does the UI work as expected?** There should be no Javascript errors in the console; all resources should load. There should be no unexpected errors. Deliberately try to break the feature to find out if there are corner cases that are not handled.

#### Code

- [ ] **Do you fully understand the introduced changes to the code?** If not ask for clarification, it might uncover ways to solve a problem in a more elegant and efficient way.
- [ ] **Does the PR introduce any inefficient database requests?** Use the debug server to check for duplicate requests.
- [ ] **Are all necessary strings marked for translation?** All strings that are exposed to users via the UI must be [marked for translation](https://docs.djangoproject.com/en/1.10/topics/i18n/translation/).

#### Tests

- [ ] **Are there sufficient test cases?** Ensure that all components are tested individually; models, forms, and serializers should be tested in isolation even if a test for a view covers these components.
- [ ] **If this is a bug fix, are tests for the issue in place?**  There must be a test case for the bug to ensure the issue won’t regress. Make sure that the tests break without the new code to fix the issue.
- [ ] **If this is a new feature or a significant change to an existing feature?** has the manual testing spreadsheet been updated with instructions for manual testing?

#### Security

- [ ] **Confirm this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets.**
- [ ] **Are all UI and API inputs run through forms or serializers?**
- [ ] **Are all external inputs validated and sanitized appropriately?**
- [ ] **Does all branching logic have a default case?**
- [ ] **Does this solution handle outliers and edge cases gracefully?**
- [ ] **Are all external communications secured and restricted to SSL?**

#### Documentation

- [ ] **Are changes to the UI documented in the platform docs?** If this PR introduces new platform site functionality or changes existing ones, the changes should be documented.
- [ ] **Are changes to the API documented in the API docs?** If this PR introduces new API functionality or changes existing ones, the changes must be documented.
- [ ] **Are reusable components documented?** If this PR introduces components that are relevant to other developers (for instance a mixin for a view or a generic form) they should be documented in the Wiki.

### CODE REVIEW

File: components/application-mgt/org.wso2.carbon.identity.application.common/src/main/java/org/wso2/carbon/identity/application/common/model/ServiceProvider.java [hunk 1]
```diff
@@ -514,5 +520,13 @@ public void setManagementApp(boolean managementApp) {
 
         isManagementApp = managementApp;
     }
+
+    public boolean isB2BSelfServiceApp() {
```
Comment (thanujalk) [inline components/application-mgt/org.wso2.carbon.identity.application.common/src/main/java/org/wso2/carbon/identity/application/common/model/ServiceProvider.java:524]: Add an empty line after the method declaration. Do the same for others.

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationManagementServiceImpl.java [hunk 2]
```diff
@@ -650,6 +650,11 @@ public void updateApplication(ServiceProvider serviceProvider, String tenantDoma
                     .isManagementApp();
             serviceProvider.setManagementApp(isManagementApp);
 
+            // Updating the isB2BSelfService flag of application is blocked, thus set it to stored value
+            boolean isB2BSelServiceApp = appDAO.getApplication(serviceProvider.getApplicationID())
```
Comment (thanujalk) [inline components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationManagementServiceImpl.java:654]: Unnecessary variable  `isB2BSelServiceApp`

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/ApplicationDAOImpl.java [hunk 3]
```diff
@@ -4710,6 +4729,15 @@ private ServiceProviderProperty buildIsManagementAppProperty(ServiceProvider sp)
         return isManagementAppProperty;
     }
 
+    private ServiceProviderProperty buildIsB2BSSAppPropertyProperty(ServiceProvider sp) {
```
Comment (piyarathnalakmali) [inline components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/ApplicationDAOImpl.java:4732]: Does having `Property` twice in the method name intended?

Comment (inthirakumaaran) [reply components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/ApplicationDAOImpl.java]: Nop, refactored the methodname

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationMgtUtil.java [hunk 4]
```diff
@@ -53,7 +53,6 @@
 import org.wso2.carbon.user.core.common.AbstractUserStoreManager;
 import org.wso2.carbon.user.core.util.UserCoreUtil;
 
-import java.util.ArrayList;
```
Comment (piyarathnalakmali) [inline components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationMgtUtil.java:56]: PR builder is failing due to missing this import

Comment (inthirakumaaran) [reply components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationMgtUtil.java]: Fixed

File: components/application-mgt/org.wso2.carbon.identity.application.common/src/main/java/org/wso2/carbon/identity/application/common/model/ServiceProvider.java [hunk 5]
```diff
@@ -514,5 +520,15 @@ public void setManagementApp(boolean managementApp) {
...
+
+        return isB2BSelfServiceApp;
+    }
+
+    public void setB2BSelfServiceApp(boolean b2BSelfServiceApp) {
```
Comment (AnuradhaSK) [inline components/application-mgt/org.wso2.carbon.identity.application.common/src/main/java/org/wso2/carbon/identity/application/common/model/ServiceProvider.java:529]:
```suggestion
    public void setB2BSelfServiceApp(boolean b2bSelfServiceApp) {
```
Is `b2BSelfServiceApp` an auto-generated variable name? for me it's difficult to read. `b2bSelfServiceApp` would give  better readability. For abbreviation of 3 letters or less we can follow the same case right

File: components/application-mgt/org.wso2.carbon.identity.application.common/src/main/java/org/wso2/carbon/identity/application/common/model/ServiceProvider.java [hunk 6]
```diff
@@ -514,5 +520,15 @@ public void setManagementApp(boolean managementApp) {
...
+
+        isB2BSelfServiceApp = b2BSelfServiceApp;
+    }
```
Comment (thanujalk) [inline components/application-mgt/org.wso2.carbon.identity.application.common/src/main/java/org/wso2/carbon/identity/application/common/model/ServiceProvider.java:532]: Can't we have the same variable name?
```suggestion
    public void setB2BSelfServiceApp(boolean isB2BSelfServiceApp) {

        this.isB2BSelfServiceApp = isB2BSelfServiceApp;
    }
```

**Review by thanujalk — DISMISSED** [review]

**Review by thanujalk — APPROVED** [review]

**Pull request merged**